engine/server/engine/enclave_manager/enclave_pool.go

```diff
@@ -47,49 +47,60 @@ func CreateEnclavePool(
 	engineVersion string,
 ) (*EnclavePool, error) {
 
+	//TODO the current implementation only removes the previous idle enclave, it's pending to implement the reusable feature
+	//TODO the reuse logic is not enable yet because we ned to store the APIC version on the APIContainer object in container-engine-lib
+	//TODO in order to using it for comparing it with the expected version
+
+	// iterate on all the existing enclaves in order to find idle enclaves already created
+	// and reuse or destroy them if these were created from old Kurtosis version
+	if err := destroyIdleEnclaves(kurtosisBackend); err != nil {
```

Comment:
I moved this before this condition `if poolSize == 0 {` because it has to be executed always